mlir/lib/Dialect/OpenMP/IR/OpenMPDialect.cpp

```diff
@@ -915,6 +925,33 @@ static LogicalResult verifyMapClause(Operation *op, OperandRange mapOperands) {
       if (isa<ExitDataOp>(op) && to)
         return emitError(op->getLoc(),
                          "from, release and delete map types are permitted");
+
+      if (isa<UpdateDataOp>(op)) {
+        if (del) {
+          return emitError(op->getLoc(),
+                           "at least one of to or from map types must be "
+                           "specified, other map types are not permitted");
+        }
+
+        if (to & from) {
```

Comment:
Did you mean to use && instead of & here?

Regardless, I think we want to check for negation of exclusive or here so it should be `if(to == from)` then emit error.

This way we can get also get rid of the other if condition below.

Reply:
> Did you mean to use && instead of & here?

I used `&` to be consistent with the other checks in the same function. I also don't really like it but I didn't want to divert from the surrounding code which uses `|` instead of `||`. For `bool`s both are equivalent.

> Regardless, I think we want to check for negation of exclusive or here so it should be if(to == from) then emit error.

Wouldn't it be better to have separate error messages for each situation? This is the main reason I separated them.

Reply:
> Did you mean to use && instead of & here?

My eyes are playing tricks on me, I used `&` to be consistent with the other checks in the same function; I thought other checks in the function use the same as well. For `bool`s both are equivalent. Anyway, changed it to `&&`.

> Regardless, I think we want to check for negation of exclusive or here so it should be if(to == from) then emit error.

Wouldn't it be better to have separate error messages for each situation? This is the main reason I separated them.

Reply:
These error messages would usually only be seen by working on MLIR and not end users of OpenMP, so usually we don't care about being too descriptive. However, it doesn't hurt, so you can leave them both in there if you want 😁